src/views/directMessages/containers/index.js

```diff
@@ -130,11 +140,30 @@ class DirectMessages extends React.Component<Props, State> {
           messageComposer={!isComposing && !isViewingThread}
         />
         <MessagesList isViewingThread={isViewingThread || isComposing}>
-          <Link to="/messages/new" onClick={() => this.setActiveThread('new')}>
-            <ComposeHeader>
+          <ComposeHeader>
+            <TabsContainer>
+              <TabsLink
+                to="/messages"
+                onClick={() => this.setActiveThread('new')}
+                isActive={isInActiveMessages}
+              >
+                Inbox
+              </TabsLink>
+              <TabsLink
+                to="/messages/archived"
+                onClick={() => this.setActiveThread('new')}
+                isActive={isInArchivedMessages}
+              >
+                Archived
+              </TabsLink>
+            </TabsContainer>
+            <NewMessageLink
+              to="/messages/new"
+              onClick={() => this.setActiveThread('new')}
```

Comment:
you are creating three times arrow function which basically do the same... it would be better to declare it as a class property, and you will have only one place where the logic is...